docs/source/architecture/sub_stakes.rst

```diff
@@ -0,0 +1,270 @@
+
+                 Before             
+
+            stake                 
+            ^                     
+            |                     
+         900+----------+          
+            |    1st   |  period
+            +----------+--->   
+            0          5          
+			
+			
+			
+                 After             
+
+            stake                 
+            ^                     
+            |                     
+         900+-------------+        
+            |      1st    |  period
+            +-------------+--->   
+            0             7     
+			
+			
+Sub-stake division
+^^^^^^^^^^^^^^^^^^
+*Methods* : ``divideStake(uint256,uint256,uint16)``
+
+If necessary, staker can extend the locking duration only for the part of tokens in a sub-stake (``divideStake()``). Contract method splits desirable sub-stake into two elements: the old sub-stake with changed locked amount and the new sub-stake. New element has the extended locking duration (by the specified number of periods) and the same start period as the old one has. Locked amount in the new sub-stake equals to requested input parameter.
+
+**Example:**
+
+    A staker divides sub-stake and extends locking time for 300 tokens for 2 additional periods:
+		- Before: 
+			* 1st sub-stake = 900 tokens for 5 periods
+		- After: 
+			* 1st sub-stake = 600 tokens for 5 periods
+			* 2nd sub-stake = 300 tokens for 7 periods
+
+    .. aafig::
+        :proportional:
+        :textual:
+
+                 Before             
+
+            stake                 
+            ^                     
+            |                     
+         900+----------+          
+            |          |  
+            |    1st   |
+            |          |  period 
+            +----------+--->   
+            0          5          
+			
+			
+			
+                 After             
+
+            stake                 
+            ^                     
+            |                     
+         900+----------+       
+            |          |
+            |    1st   |
+         300+----------+--+
+            |      2nd    |  period
+            +-------------+--->   
+            0          5  7     
+
+
+Slashing
+^^^^^^^^
+*See:* `the slashing protocol`_ 
+
+.. _`the slashing protocol`: https://docs.nucypher.com/en/latest/architecture/slashing.html
+
+
+
+
+Flags that affect the sub-stake array
+-------------------------------------
+
+Re-staking
+^^^^^^^^^^
+*Used in methods* : ``confirmActivity()``, ``mint()``
+
+When re-staking is turned off, the number of locked tokens in sub-stakes does not change by itself.
+In case when re-staking parameter is on then all staking reward is locked as the part of each involved sub-stake (inside ``confirmActivity()`` and/or ``mint()``).  Accordingly, each such sub-stake has increased locked amount (by reward) and the number of sub-stakes stays unchanged.
+
+**Example:**
+
+    A staker has few sub-stakes and calls ``mint()``:
```

Comment:
For this scenario, is it the case that 1st sub-stake produces 50 tokens and 2nd sub-stake produces 20 tokens as rewards? If so, we should state that explicitly for the scenario.

Reply:
Yep, you're totally right, where to put it? in description of before? Or in "after" sections?

Reply:
I took your suggestion with some modifications, tell me what do you think

Reply:
👍 added minor comment about tense.